### ISSUE TITLE
Periodic Error when Abandoning a Quest (Race Condition)

### ISSUE THREAD

**Issue by Nillx** [butter-cookie-kitkat/ButterQuestTracker#50]
Noticed that im getting lots of errors again. Not sure what is causing it.

```
38x ...tterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:76: attempt to index local 'quest' (a nil value)
...tterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:76: in function `SetIconsVisibility'
...tterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:42: in function `listener'
...erQuestTracker\QuestWatchHelper\QuestWatchHelper.lua:28: in function <...erQuestTracker\QuestWatchHelper\QuestWatchHelper.lua:26>
SharedXML\C_TimerAugment.lua:16: in function <SharedXML\C_TimerAugment.lua:14>

Locals:
self = <table> {
 autoHideIcons = true
 GetDestinationCoordinates = <function> defined @ButterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:82
 IsSupported = <function> defined @ButterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:35
 SetIconsVisibility = <function> defined @ButterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:62
 GetDistanceToClosestObjective = <function> defined @ButterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:164
 SetAutoHideQuestHelperIcons = <function> defined @ButterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:46
}
questID = nil
visible = false
quest = nil
(*temporary) = true
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'quest' (a nil value)"
QLH = <table> {
 OnQuestUpdated = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:73
 SetQuestsLastUpdated = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:77
 GetQuestSummary = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:348
 ToggleQuest = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:136
 _questFrame = QuestLogFrame {
 }
 GetDifficulty = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:155
 GetQuestFrame = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:108
 RevertSelection = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:362
 AbandonQuest = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:334
 GetIndexFromQuestID = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:102
 ShareQuest = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:341
 Select = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:357
 Refresh = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:186
 IsShown = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:128
 GetDifficultyColor = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:171
 IsQuestSharable = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:326
 GetObjectives = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:309
 GetQuests = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:276
 GetWatchedQuests = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:296
 GetQuestIDFromIndex = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:92
 GetQuestCount = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:292
 IsQuestSelected = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:132
 GetQuest = <function> defined @ButterQuestTracker\QuestLogHelper\QuestLogHelper.lua:284
}
refresh = <function> defined @ButterQuestTracker\LibQuestHelpers\LibQuestHelpers.lua:15
```

**Comment by cecilia-sanare** [comment]
Looks like you have sorting by quest proximity enabled and auto hiding icons?

**Comment by cecilia-sanare** [comment]
Actually nevermind looks like you just have auto hiding icons enabled.

Also did you abandon a quest right before this happened?

**Comment by Nillx** [comment]
I did abandoned a quest earlier today yes. Also installed this addon (scanning for enemy players around me) https://www.curseforge.com/wow/addons/spy-classic

**Comment by cecilia-sanare** [comment]
Hmm, did it happen upon logging in or something?

**Comment by Nillx** [comment]
i took an escort quest and abandoned it, then relogged

**Comment by cecilia-sanare** [comment]
Hey @Nillx I just released 1.9.0-rc.3 that contains a fix for this issue.

Let me know if you run into anything else!

**Comment by Nillx** [comment]
Thanks, so far so good.
